etl/stage/aneel/base.py

```
@@ -0,0 +1,152 @@
+
+    def run(self) -> None:
+        """Executa o ciclo completo de extração e carregamento de dados.
+
+        Este é o método principal que orquestra todo o processo EL.
+        Ele primeiro chama `before()` para configurar, depois itera sobre
+        todas as páginas de dados fornecidas pelo objeto `search`, chamando
+        `extract()` para cada página e, em seguida, `load()` para persistir
+        os dados extraídos.
+        """
+        self.before()
+        for self.page in self.search.iterpage():
```

Ruff (0.12.2)

Loop control variable `self` overrides iterable it iterates

(B020)